docs/topics/native/native-ios-integration.md

```diff
@@ -244,6 +245,26 @@ It requires two GC cycles to collect these four objects because deinitialization
 after the GC cycle. The limitation stems from `deinit`, which can call arbitrary code, including the Kotlin code that
 cannot be run during the GC pause.
 
+### Retain cycles
+
+In a retain cycle, a number of objects refer each other using strong references cyclically:
+
+![Retain cycles](native-retain-cycle.png){height=200}
+
+Kotlin's GC and Objective-C's ARC handle retain cycles differently. When objects become unreachable, the GC can properly
+reclaim such cycles, while Objective-C's ARC can't. Therefore, retain cycles of Kotlin objects can be reclaimed,
+while retain cycles of Swift/Objective-C objects cannot.
+
+Consider the case when retain cycles contain both Objective-C and Kotlin objects:
+
+![Retain cycles with Objective-C and Kotlin objects](native-objc-kotlin-retain-cycles.png){height=150}
+
+This involves combining Kotlin's and Objective-C's memory management models that cannot handle (reclaim) retain cycles
+together. That means that if at least one Objective-C object is present, the retain cycle of a whole graph of objects
+cannot be reclaimed, and it's impossible to break the cycle from the Kotlin side.
+
+Unfortunately, no special instruments are currently available to detect retain cycles in Kotlin/Native code.
```

Comment:
Maybe we can suggest to use the same techniques as in Swift to break retain cycles? For example, https://docs.swift.org/swift-book/documentation/the-swift-programming-language/automaticreferencecounting/#Weak-References or https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/mmPractical.html#//apple_ref/doc/uid/TP40004447-1000810

Reply:
Thank you! Both links describe weak references, I chose a more general section from the first article